doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/user-guides/kuberay-vs-legacy.md

```diff
@@ -2,23 +2,115 @@
+The two operators differ primarily in internal design and implementation.
+There are also some differences in configuration details.
 
-Internal design, choice of language and framework, overall stability and scalability.
-Interface details.
+### Implementation and architecture
+**Legacy Ray Operator** The legacy Ray Operator is implemented in Python.
+Kubernetes event handling is implemented using the [Kopf](https://kopf.readthedocs.io/en/stable/) framework.
+The operator invokes Ray cluster launcher and autoscaler code to manage Ray clusters.
+The operator forks an autoscaler subprocess for each Ray cluster it manages.
+The Ray autoscaler subprocesses create and delete Ray pods directly.
+
+**KubeRay Operator** The KubeRay operator is implemented in Golang using standard tools
+for building Kubernetes operators, including the [KubeBuilder](https://github.com/kubernetes-sigs/kubebuilder)
+operator framework
+and the [client-go](https://github.com/kubernetes/client-go) client library.
+The KubeRay operator is structurally simpler than the legacy Ray Operator;
+rather than running many Ray autoscalers in subprocesses, the KubeRay operator implements a simple
+reconciliation loop. The reconciliation loop creates and deletes Ray pods to match the desired
+state expressed in each RayCluster CR.
+Each Ray cluster runs its own autoscaler as a sidecar to the Ray head pod.
+The Ray autoscaler communicates desired scale to the KubeRay operator by writing to the RayCluster
+custom resource.
+
+### Scalability
+The KubeRay operator is more scalable than the legacy Ray Operator. Specifically, the
+KubeRay operator can simultaneously manage more Ray clusters.
+
+**Legacy Ray Operator** Each Ray autoscaler consumes nontrivial memory and CPU resources.
+Since the legacy Ray Operator runs many autoscalers in one pod, it cannot manage many Ray clusters.
+
+**KubeRay Operator** The KubeRay operator does not run Ray autoscaler processes.
+Each Ray autoscaler runs as a sidecar to the Ray head. Since managing each Ray cluster is cheap,
+the KubeRay operator can manage many Ray clusters.
+
+### Ray version compatibility
+
+**Legacy Ray Operator**
+It is recommended to use the same Ray version in the legacy Ray operator
+as in all of the Ray pods managed by the operator.
+Matching Ray versions is required to maintain compatibility between autoscaler code
+running in the operator pod and Ray code running in the Ray cluster.
+
+**KubeRay Operator**
+The KubeRay operator is compatible with many Ray versions.
+Compatibility of KubeRay v0.3.0 with Ray versions 1.11, 1.12, 1.13, and 2.0 is tested explicitly.
+
+Note however that autoscaling with KubeRay is supported only with Ray versions
+at least as new as 1.11.0.
+
+### Configuration details
+Some details of Ray cluster configuration are different with KubeRay; see the next section
+for migration notes. Refer to the {ref}`configuration guide<kuberay-config>` for comprehensive
+information.
 
 ## Migration notes
```

Comment:
Since many people will have this problem, we need to be a bit more explicit here. Let's have a worked out example here where we translate an old complete example yaml into the new format and show the diff in a way that is well consumable in the docs :)

Reply:
(this is very important I think)

Reply:
Sure, I'll do a conversion of Helm for old operator to CR Yaml for new operator
(since Helm is the most prominently documented way for the old one)

Reply:
Added, will go back over to edit.